core/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -120,8 +121,11 @@ protected boolean validate(ClusterState newState) {
         AsyncSingleAction(Task task, Request request, ActionListener<Response> listener) {
             this.task = task;
             this.request = request;
-            if (task != null) {
-                request.setParentTask(clusterService.localNode().getId(), task.getId());
+            DiscoveryNode discoveryNode = clusterService.localNode();
+            if (task != null && discoveryNode != null) {
+                request.setParentTask(discoveryNode.getId(), task.getId());
+            } else {
+                throw new IllegalStateException("No local node to execute task on. Did you forget to start the node?");
```

Comment:
I think this should be handled more broadly, probably in the cluster service itself because this is definitely not the only place in the code base that will expect the local node to not be null.